modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js

```diff
@@ -165,6 +169,81 @@ const CustomViewsControls = () => {
 		});
 	};
 
+	const renameActiveCustomView = ({label, processClose}) => {
+		const url = new URL(
+			`${appURL}/fds/${fdsName}/custom-views/${activeCustomViewId}/label`
+		);
+
+		url.searchParams.append('portletId', portletId);
+
+		fetch(url, {
+			body: objectToFormData({
+				customViewLabel: label,
+			}),
+			method: 'POST',
+		})
+			.then((response) => {
+				if (response.ok) {
+					if (processClose) {
+						processClose();
+					}
+
+					openToast({
+						message: Liferay.Language.get(
+							'view-was-renamed-successfully'
+						),
+						type: 'success',
+					});
+
+					viewsDispatch({
+						type: VIEWS_ACTION_TYPES.RENAME_ACTIVE_CUSTOM_VIEW,
+						value: {
+							label,
+						},
+					});
+				}
+				else {
+					openToast({
+						message: Liferay.Language.get(
+							'an-unexpected-error-occurred'
```

Comment:
I'm just keeping this in mind now since we're currently facing a11y bugs: displaying [descriptive error messages](https://accessibility.huit.harvard.edu/provide-helpful-error-messages) is also an accessibility concern. Maybe we could handle this now so it doesn't come up in the future.

Is there an error message in the response?

Reply:
That's a good call, we can definitely improve error reporting. @LuismiBarcos already suggested something similar in https://github.com/liferay-frontend/liferay-portal/pull/2743#discussion_r979813723.

One way to do it is to have unique error messages for different error codes from REST responses. For example, a 404 on `@GET` on custom views could say something like "Unable to find view." To make this work, we first need to extend backend. Currently, not counting this PR, we only wrap everything in `try/catch` and throw `INTERNAL_SERVER_ERROR`. With this, "unexpected error" in frontend is, unfortunately, quite fitting :sweat_smile:    

@dsanz @pablo-agulla Improving custom views error reporting is a potential future story.